packages/shorebird_cli/lib/src/shorebird_process.dart

```diff
@@ -26,42 +30,89 @@ abstract class ShorebirdProcess {
     String executable,
     List<String> arguments, {
     bool runInShell = false,
+    Map<String, String>? environment,
     String? workingDirectory,
+    bool useVendedFlutter = true,
   }) {
+    final resolvedEnvironment = environment ?? {};
+    if (useVendedFlutter) {
+      resolvedEnvironment.addAll(
```

Comment:
I might have left myself a comment that addAll overwrites (I had to look it up): https://api.flutter.dev/flutter/dart-core/Map/addAll.html 🤷

Reply:
Done.